### ISSUE TITLE
Docs ansible-1.1: "register" example broken?

### ISSUE THREAD

**Issue by timlnx** [ansible/ansible#3099]
```
[root@util01 ansible]# ansible-playbook --version
ansible-playbook 1.1
[root@util01 ansible]# rpm -q ansible
ansible-1.1-1.el6.noarch
[root@util01 ansible]# rpm -q --qf "%{NAME} - %{VENDOR}\n" ansible
ansible - Fedora Project
[root@util01 ansible]# which ansible-playbook
alias ansible-playbook='ansible-playbook -v -i /root/ansible/hosts'
        /usr/bin/ansible-playbook
```

Description:
The example [1] for `register` appears to be broken.

Steps to reproduce:
1. Copy example from [1] into a yaml file
2. Run the file with ansible-playbook

```
[root@util01 tmp]# grep "hi" /etc/motd
This motd has the word 'hi' in it
[root@util01 tmp]# cat test_when.yaml
- name: test play
  hosts: util

  tasks:

      - action: shell cat /etc/motd
        register: motd_contents

      - action: shell echo "motd contains the word hi"
        when: motd_contents.stdout.find('hi') != -1
```

Expected Result:
Ansible prints the message "motd contains the word hi"

Observed result:

```
[root@util01 tmp]# ansible-playbook test_when.yaml
ERROR: invalid usage of when_ operator: motd_contents.stdout.find('hi') != -1
```

This is failing when it gets to the final 'else' clause in `compile_when_to_only_if`. I added some debugging statements:

```
    else:
        print "Tokens: " + str(tokens)
        print map(lambda t: type(t), tokens)
        raise errors.AnsibleError("invalid usage of when_ operator: %s" % expression)
```

which outputs:

```
[root@util01 tmp]# ansible-playbook test_when.yaml
Tokens: ["motd_contents.stdout.find('hi')", '!=', '-1']
[<type 'str'>, <type 'str'>, <type 'str'>]
ERROR: invalid usage of when_ operator: motd_contents.stdout.find('hi') != -1
```

[1] http://ansible.cc/docs/playbooks2.html#register-variables

**Comment by mpdehaan** [comment]
Yeah, that isn't the Ansible 1.1 docs, it is the 1.2 docs, which is somewhat unclearly mentioned at the top of the docs page but becomes a moot point in 7 days when 1.2 comes out.

You want this:

http://ansible.cc/docs/released/1.1/

We wanted to stop talking about the old ways so they would be used less, rather than saying there are N confusing ways to do this, but this is the one you should use, we just talk about the new cleaned up ways.

**Comment by timlnx** [comment]
OK. That probably explains a lot. Thanks! I'll try this out now.

On Sun, Jun 2, 2013 at 5:20 PM, Michael DeHaan notifications@github.comwrote:

> Yeah, that isn't the Ansible 1.1 docs, it is the 1.2 docs, which is
> somewhat unclearly mentioned at the top of the docs page but becomes a moot
> point in 7 days when 1.2 comes out.
> 
> You want this:
> 
> http://ansible.cc/docs/released/1.1/
> 
> We wanted to stop talking about the old ways so they would be used less,
> rather than saying there are N confusing ways to do this, but this is the
> one you should use, we just talk about the new cleaned up ways.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/ansible/ansible/issues/3099#issuecomment-18814122
> .

## 

--Tim Bielawa
http://blog.lnx.cx
